File: php/features.php
<?php
declare(strict_types=1);

// BerufsbildungDatenExporte SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BerufsbildungDatenExporteFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BerufsbildungDatenExporteBaseFeature();
            case "test":
                return new BerufsbildungDatenExporteTestFeature();
            default:
                return new BerufsbildungDatenExporteBaseFeature();
        }
    }
}

File: py/berufsbildungdatenexporte_sdk/features.py
# BerufsbildungDatenExporte SDK feature factory

from berufsbildungdatenexporte_sdk.feature.base_feature import BerufsbildungDatenExporteBaseFeature
from berufsbildungdatenexporte_sdk.feature.test_feature import BerufsbildungDatenExporteTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BerufsbildungDatenExporteBaseFeature(),
        "test": lambda: BerufsbildungDatenExporteTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
